src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -114,6 +120,11 @@ public void onConnectionResumed(boolean sessionPresent) {
         this.ses = ses;
     }
 
+    long getThrottlingWaitTimeMicros() {
```

Comment:
The bandwidth limiter needs one permit per byte, right? Don't we need to take message size into account here?

Reply:
The time to wait is independent of how much you want to send, that just further delays the next time that you can take tokens out of the bucket. Basically the initial request isn't delayed by how big it is, but subsequent requests will be.

Reply:
Nit: prolly worth calling out in a comment just so others don't also assume this is a bug